airmozilla/starred/tests/test_starred.py

```diff
@@ -0,0 +1,198 @@
+import json
+
+from django.test import TestCase
+from django.contrib.auth.models import User
+
+from funfactory.urlresolvers import reverse
+from nose.tools import eq_, ok_
+
+from airmozilla.main.models import Event
+from airmozilla.starred.models import (
+    StarredEvents,
+)
+
+
+class TestStarredEvent(TestCase):
+    fixtures = ['airmozilla/manage/tests/main_testdata.json']
+# test that a starred event can be synced
+# test that an unstarred event can be synced
+# test that unchanged events remain that way
+# test that unknown event id does the right thing (whatever that is)
+# test that the proper list is returned from a get
+# test that the proper list of starred events is displayed
+# add star, sync 
+# remove star, add different star, sync
+
+# test what happens when a user isn't logged in
+
+
+    def setUp(self):
+        super(TestStarredEvent, self).setUp()
+
+        # create the url
+        self.url = reverse('starred:sync_starred_events')
+        # so, let's sign in
+        User.objects.create_user('lisa', password='secret')
+        assert self.client.login(username='lisa', password='secret')
+
+        # instantiate test event
+        event = Event.objects.get(title='Test event')
+        # create more events 
+        Event.objects.create(
+            title='A Good Event',
+            slug='event1',
+            description=event.description,
+            start_time=event.start_time,
+            privacy=Event.PRIVACY_PUBLIC,
+            placeholder_img=event.placeholder_img,
+            location=event.location,
+        )
+        Event.objects.create(
+            title='A Great Event',
+            slug='event2',
+            description=event.description,
+            start_time=event.start_time,
+            privacy=Event.PRIVACY_PUBLIC,
+            placeholder_img=event.placeholder_img,
+            location=event.location,
+        )
+        Event.objects.create(
+            title='The Best Event',
+            slug='event3',
+            description=event.description,
+            start_time=event.start_time,
+            privacy=Event.PRIVACY_PUBLIC,
+            placeholder_img=event.placeholder_img,
+            location=event.location,
+        )
+        # get events from the database
+        self.events = Event.objects.all()
```

Comment:
This whole setUp is getting too big and creating too much stuff. Fixtures are tricky but basically they should be kept at a very small minimum. The `'Test event'` sample event that is loaded from `airmozilla/manage/tests/main_testdata.json` is used so incredibly much that it was worth it so it's an exception. 

Basically, if there's something in here that isn't needed for every (or 95%) test in the class, then it's excess. 

The reasoning is twofold:

1. For someone reading the test, it's better to be explicit (try typing `import this` in any python interpreter shell) about what data is available so as to remove any guesswork. If you can see the `Event.objects.create(...)` about 5 lines away from where you're reading code it's easier to comprehend what's going on.

2. One day we might remove one of the tests but in the rush miss to clean up the necessary fixture that was used by only that one last test. At that point it's really confusing why the fixture is needed. And taking the time to do clean-up refactoring is rare. This is the same predicament when you add stuff to a .css file needed by some widget and much later in time you delete the widget but fail to clean up the .css file and then that file just grows and grows. 

Does that make sense?